### ISSUE TITLE
Add optional UMI handling

### ISSUE THREAD

**Issue by ewels** [SciLifeLab/NGI-RNAseq#145]
It would be nice to add optional handling of UMIs, probably using this project: https://github.com/CGATOxford/UMI-tools

**Comment by TomSmithCGAT** [comment]
Hi @ewels - Let us know how you get on integrating UMI-tools into your pipeline. Very happy to help out via GitHub issues if you have any questions/issues etc

If you haven't already seen it, we have just released a quick scRNA-Seq tutorial: https://github.com/CGATOxford/UMI-tools/blob/master/doc/Single_cell_tutorial.md

**Comment by ewels** [comment]
Thanks @TomSmithCGAT! Yes I was just reading over it this morning. I particularly like the _TL;DR;_ section at the top 😉 

Whilst the single-cell / 10X analysis looks great, we're not currently running single cell projects in our facility so I think that we should probably keep things simple and just run `umi_tools extract` and `umi_tools dedup` if the pipeline is run with `--bc-pattern`. To be honest, my gut feeling is that single-cell data is different enough to warrant a totally separate pipeline anyway.

Phil

**Comment by ewels** [comment]
ps. How did you find this issue so quickly..??

**Comment by TomSmithCGAT** [comment]
Oh yeah, I'd assumed you were wanting to support UMIs for bulk RNA-Seq. Agree a separate pipeline is the best approach for scRNA-Seq. 

I sporadically check who the new github stars are from (the speed was just by chance!). I saw you were at SciLife so I searched for "UMI-tools" in the organisation's github just to see what you were using UMI-tools for as we don't really have a very good idea what our user base actually is. We only really hear from people who have issues/questions etc - hopefully that's not everyone!
